Source/Scene/Model.js

```diff
@@ -1186,8 +1187,21 @@ define([
         var cacheKey = defaultValue(options.cacheKey, getAbsoluteUri(url));
 
         options = clone(options);
-        options.basePath = getBaseUri(url, true);
-        options.cacheKey = cacheKey;
+        var providedBasePath = options.basePath;
+        if (!defined(options.basePath)) {
+            options.basePath = getBaseUri(url, true);
+        }
+        if (!defined(options.cacheKey)) {
+            if (defined(providedBasePath)) {
+                options.cacheKey = cacheKey + options.basePath;
```

Comment:
If these are both URLs, we should use the [joinUrls](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/joinUrls.js) function to concat them